dashboard/lib/pd/foorm/workshop_summarizer.rb

```diff
@@ -0,0 +1,93 @@
+# Summarizes a set of survey responses by counting number of responses or getting lists of free text
+# responses
+module Pd::Foorm
+  class WorkshopSummarizer
+    include Constants
+    extend Helper
+
+    # result format:
+    # {
+    #   <form_name>.<form_version> =>
+    #   {
+    #     num_respondents: 4,
+    #     <question1-name> => { num_respondents: 3, <answer1-name>: 5, <answer2-name>: 3, other_answers: ["other 1", "other 2"]...},
+    #     <question2-name> => {<answer1-name>: 5, <answer2-name>: 3,...},
+    #     <question3-name> => ['abc', 'def']
+    #   }
+    # }
+    # Where the value for a question name is an array answers for a text question
+    # or a summary of answer choices for select/matrix
+    # questions. If question is a matrix responses will be nested like so:
+    # {<matrix-name> => {<question1-name: {<answer1-name>: 2, <answer2-name>: 3}}}
+    # num_respondents within the hash is used only for multi-select.
+    # other_answers is used if the select question has a free text other field
+    def self.summarize_answers_by_survey(foorm_submissions, parsed_forms, ws_submissions)
+      workshop_summary = {}
+      foorm_submissions.each do |submission|
+        form_key = get_form_key(submission.form_name, submission.form_version)
+        survey_key = get_survey_key(ws_submissions.where(foorm_submission_id: submission.id).first)
+        next unless parsed_forms[form_key] && parsed_forms[form_key]
+        form_questions = parsed_forms[form_key]
+        workshop_summary[survey_key] ||= {response_count: 0}
+        workshop_summary[survey_key][:response_count] += 1
+        current_workshop_summary = workshop_summary[survey_key][form_key] || {}
+        workshop_summary[survey_key][form_key] = add_single_submission_to_summary(submission, current_workshop_summary, form_questions)
```

Comment:
Does this match the output structure in the above function description?

Reply:
yes